File: boilerplateBE/src/Starter.Application/Features/Auth/DTOs/LoginResponseDto.cs
using Starter.Application.Features.Users.DTOs;

namespace Starter.Application.Features.Auth.DTOs;

public sealed record LoginResponseDto(
    string? AccessToken,
    string? RefreshToken,
    DateTime? ExpiresAt,
    UserDto? User,
    bool RequiresTwoFactor = false);

File: boilerplateMobile/lib/core/features/auth/data/dtos/login_response_dto.dart
import 'package:boilerplate_mobile/core/features/auth/data/dtos/user_dto.dart';
import 'package:freezed_annotation/freezed_annotation.dart';

part 'login_response_dto.freezed.dart';
part 'login_response_dto.g.dart';

@freezed
abstract class LoginResponseDto with _$LoginResponseDto {
  const factory LoginResponseDto({
    String? accessToken,
    String? refreshToken,
    String? expiresAt,
    UserDto? user,
    @Default(false) bool requiresTwoFactor,
  }) = _LoginResponseDto;

  factory LoginResponseDto.fromJson(Map<String, dynamic> json) =>
      _$LoginResponseDtoFromJson(json);
}
